### ISSUE TITLE
Improve Debugging for template errors

### ISSUE THREAD

**Issue by rhmms** [helm/helm#6936]
Currently if a helm template generates invalid yaml you get cryptic error messages like this: 

```
Error: YAML parse error on mysql/templates/deployment.yaml: error converting YAML to JSON: yaml: line 106: mapping values are not allowed in this context
```

When trying to debug this, there is currently no easy way to do this, unless you printline debug your templates. This is not nice. In this example the line number 106 reflects the line in the rendered template, which is not visible to the user at all. This should be improved. 

A simple way to improve this, would be to print the rendered templates in debug mode (right after this https://github.com/helm/helm/blob/865c46c014cdb7622f97ae287ee92fb8a280f3a9/pkg/action/install.go#L427,L431)

This should include line numbers, to easy debugging.

**Comment by rhmms** [comment]
Example code:

```
diff --git a/pkg/action/install.go b/pkg/action/install.go
index 8f1b5528..56c96811 100644
--- a/pkg/action/install.go
+++ b/pkg/action/install.go
@@ -19,6 +19,7 @@ package action
 import (
        "bytes"
        "fmt"
+       "bufio"
        "io/ioutil"
        "os"
        "path"
@@ -429,6 +430,24 @@ func (c *Configuration) renderResources(ch *chart.Chart, values chartutil.Values
                return hs, b, "", err
        }

+       // If the template is broken and generates non yaml files (e.g. by skipping line endings etc)
+       // you get an error message by that there is an error in line xxx of the file deployment.yaml.
+       // However the line number is the line number of the generated yaml "text" after the template
+       // engine, which is rather stupid and does not know yaml at all. Debugging those issues
+       // is hard, therefore we print the rendered templates with line numbers in debug mode.
+       c.Log("Result after template processor\n--------------")
+       for fName, f := range files {
+               scanner := bufio.NewScanner(strings.NewReader(f))
+               scanner.Split(bufio.ScanLines)
+               c.Log(fmt.Sprintf("#Source:%s\n---", fName))
+               count := 0
+               for scanner.Scan() {
+                       count++
+                       c.Log(fmt.Sprintf("%d\t%s",count,scanner.Text()))
+               }
+       }
+
+
        // NOTES.txt gets rendered like all the other files, but because it's not a hook nor a resource,
        // pull it out of here into a separate file so that we can actually use the output of the rendered
        // text file. We have to spin through this map because the file contains path information, so we
```

**Comment by rhmms** [comment]
This is even more important if you are using the "tpl" function to interpolate values.yaml.

**Comment by joaocc** [comment]
Any news on this?
Besides the suggestion above, it would also be very useful to be able to have the option to render the intermediate YAML that is being converted to JSON.
Thx

**Comment by bacongobbler** [comment]
Duplicate of #7088, which was fixed with #7556.
